src/controllers/UserController.ts

```diff
@@ -0,0 +1,27 @@
+import API, {updateUserDto, UserApi, userPasswordDto} from '../api/UserApi';
+import AuthController from './AuthController';
+
+export class UserController {
+  private readonly api: UserApi;
+
+  constructor() {
+    this.api = API;
```

Comment:
Чтобы класс меньше зависел отвнешнего кода, я бы предложил не ссылаться, на уже созданный где-то экземпляр API, а создать его тут через new или передать через параметры аргумента
```suggestion
    this.userAPI = new UserAPI();
```